### PR TITLE
Fix thread safety in Glue metastore tests

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/metastore/glue/TestGlueInputConverter.java [hunk 1]
```diff
@@ -25,21 +25,22 @@
...
 {
     private Database testDb;
     private Table testTbl;
     private Partition testPartition;
 
-    @BeforeClass
+    @BeforeMethod
     public void setup()
     {
         testDb = TestMetastoreObjects.getPrestoTestDatabase();
```
Comment (martint) [inline presto-hive/src/test/java/com/facebook/presto/hive/metastore/glue/TestGlueInputConverter.java]: Why do these need to go in a `setup()` method? Make the fields final and initialize there, instead.

Comment (electrum) [reply presto-hive/src/test/java/com/facebook/presto/hive/metastore/glue/TestGlueInputConverter.java]: The objects are mutable and are modified by the tests

File: presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java [hunk 2]
```diff
@@ -542,11 +543,11 @@ public void dropColumn(String databaseName, String tableName, String columnName)
...
 
-    private List<String> buildPartitionNames(List<Column> partitionColumns, List<Partition> partitions)
+    private static List<String> buildPartitionNames(List<Column> partitionColumns, List<Partition> partitions)
     {
-        List<String> partitionNames = new ArrayList<>();
-        partitions.stream().forEach(partition -> partitionNames.add(makePartName(partitionColumns, partition.getValues())));
-        return partitionNames;
+        return partitions.stream()
+                .map(partition -> makePartName(partitionColumns, partition.getValues()))
+                .collect(toList());
```
Comment (findepi) [inline presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java]: toImmutableList ?

Comment (electrum) [reply presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java]: Sorry, I missed your comment before merging.

Comment (findepi) [reply presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java]: no problem

**Pull request merged**